pandas/core/frame.py

```diff
@@ -718,7 +718,7 @@ def __init__(
                     "is deprecated and will raise in a future version. "
                     "Use public APIs instead.",
                     DeprecationWarning,
```

Comment:
should this become a FutureWarning too?

Reply:
Yeah probably, my intention of this PR was to just change the stacklevel considering pyarrow 15 was released. I guess it makes sense to change the warning class while I'm at it.

Reply:
I think I would prefer waiting a bit longer with changing it to FutureWarning. At the very earliest, we can only remove this in pandas 4.0 with our current guidelines, so then we still have time to switch to a FutureWarning in any of the 3.x releases.

A reason to wait a bit longer for the FutureWarning is that if a user has pandas 3.0 with pyarrow 14 (not that old at this point), a FutureWarning would bubble up to the user while a DeprecationWarning stays hidden (while any direct user of it will still see the warning, which is what we care about most)